src/components/screens/InitialScreen.tsx

```diff
@@ -29,11 +28,10 @@ const InitialScreen = ({ setIsWalletModalOpen, isWalletModalOpen }: Props) => {
   } = useSwapContext();
   const { setScreen } = useScreenState();
 
-  const walletPublicKey = useMemo(() => wallet?.adapter.publicKey?.toString(), [wallet?.adapter.publicKey]);
-
   const balance = useMemo(() => {
+    if (form.fromMint === WRAPPED_SOL_MINT.toString()) return nativeAccount?.balance || 0;
```

Comment:
Just wonder why only when it's wrapped sol we check from the native account?

Reply:
Jupiter FE, and Terminal, and many places, uses `wrapAndUnwrapSOL: true` by default
it means native SOL are wrapped into "Wrapped SOL", trade, and then unwrapped back into native SOL

and they share the same mint